config/config.go

```diff
@@ -186,26 +185,20 @@ func defaultTestConfig() BaseConfig {
 }
 
 // LoadConfig load the config file.
-func LoadConfig(fileLocation string, vip *viper.Viper) (err error) {
-	if fileLocation == "" {
-		fileLocation = defaultConfigFileName
-	}
-
-	vip.SetConfigFile(fileLocation)
-	err = vip.ReadInConfig()
-
-	if err != nil {
-		if fileLocation != defaultConfigFileName {
-			log.Warning("failed loading config from %v trying %v. error %v", fileLocation, defaultConfigFileName, err)
-			vip.SetConfigFile(defaultConfigFileName)
-			err = vip.ReadInConfig()
-		}
-		// we change err so check again
-		if err != nil {
-			return fmt.Errorf("failed to read config file %v", err)
+func LoadConfig(config string, vip *viper.Viper) error {
+	if config == defaultConfigFileName || config == "" {
+		if _, err := os.Stat(defaultConfigFileName); err != nil {
```

Comment:
i honestly don't understand the purpose of `defaultConfigFileName`, which is set to `config.toml`. this file was deleted a while back i think.

the code would be simpler without this variable, and move on (return nil) `config == ""`

Reply:
i thought that it is kind of a ux thing, but you are right there is no need for it